Sources/NIORedis/RESP/RESPEncoder.swift

```diff
@@ -11,26 +9,25 @@ public final class RESPEncoder {
     /// See https://redis.io/topics/protocol
     /// - Parameter value: The `RESPValue` to encode.
     /// - Returns: The encoded value as a collection of bytes.
-    public func encode(_ value: RESPValue) -> Data {
+    public func encode(_ value: RESPValue, into buffer: inout ByteBuffer) {
         switch value {
         case .simpleString(let string):
-            return "+\(string)\r\n".convertedToData()
-
-        case .bulkString(let data):
-            return "$\(data.count)\r\n".convertedToData() + data + "\r\n".convertedToData()
-
+            buffer.writeString("+\(string)\r\n")
```

Comment:
I wonder what the performance difference of doing this would be:
```swift
buffer.writeString("+")
buffer.writeString(string)
buffer.writeString("\r\n")
```

The difference being 3 calls to `writeString` instead of doing string interpolation (which I think  if relatively expensive)